sw/device/tests/sim_dv/spi_tx_rx_test.c

```diff
@@ -0,0 +1,391 @@
+  if (bytes_transferred != SPI_DEVICE_DATASET_SIZE) {
+    LOG_ERROR(
+        "SPI_DEVICE TX_FIFO transferred bytes mismatched: {act: %0d, exp: %0d}",
+        bytes_transferred, SPI_DEVICE_DATASET_SIZE);
+  } else {
+    LOG_INFO("Transferred %0d bytes to SPI_DEVICE TX_FIFO.", bytes_transferred);
+  }
+
+  // set tx level to half of the transfer data size (64 bytes) to test TX SRAM
+  // FIFO interrupt
+  CHECK(dif_spi_device_set_irq_levels(spi_device, rx_level, tx_level) ==
+        kDifSpiDeviceResultOk);
+  exp_spi_device_irq_tx_below_level = true;
+
+  while (!read_rx_fifo_done || !spi_device_irq_rx_full_fired ||
+         !spi_device_irq_rx_above_level_fired ||
+         !spi_device_irq_tx_below_level_fired ||
+         !spi_device_irq_tx_underflow_fired ||
+         !spi_device_irq_rx_overflow_fired) {
+    // set rx tx level back to default value to test RX SRAM FIFO interrupt
+    if (spi_device_irq_tx_below_level_fired &&
+        exp_spi_device_irq_tx_below_level) {
+      CHECK(dif_spi_device_set_irq_levels(spi_device, rx_level, 0) ==
+            kDifSpiDeviceResultOk);
+      exp_spi_device_irq_tx_below_level = false;
+      exp_spi_device_irq_rx_above_level = true;
+    }
+
+    // SPI_HOST sent 128 bytes to RX_FIFO
+    if (spi_device_irq_rx_above_level_fired &&
+        exp_spi_device_irq_rx_above_level) {
+      exp_spi_device_irq_rx_above_level = false;
+    }
+
+    // read out 128 bytes from the RX_FIFO and compare to the expected data
+    if (spi_device_irq_rx_above_level_fired && !read_rx_fifo_done) {
+      CHECK(dif_spi_device_recv(spi_device, (void *)spi_device_rx_data,
+                                SPI_DEVICE_DATASET_SIZE,
+                                &bytes_recved) == kDifSpiDeviceResultOk);
+      if (bytes_recved == SPI_DEVICE_DATASET_SIZE) {
+        LOG_INFO("Received %0d bytes from SPI_DEVICE RX_FIFO.", bytes_recved);
+        read_rx_fifo_done = true;
+      } else {
+        LOG_ERROR(
+            "SPI_DEVICE RX_FIFO received bytes mismatched: {act: %0d, exp: "
+            "%0d}",
+            bytes_recved, SPI_DEVICE_DATASET_SIZE);
+      }
+      // Check data consistency.
+      LOG_INFO("Checking the received SPI_HOST RX_FIFO data for consistency.");
+      for (int i = 0; i < SPI_DEVICE_DATASET_SIZE; ++i) {
+        CHECK(spi_device_rx_data[i] == exp_spi_device_rx_data[i],
+              "SPI_DEVICE RX_FIFO data[%0d] mismatched: {act: %x, exp: %x}", i,
+              spi_device_rx_data[i], exp_spi_device_rx_data[i]);
+      }
+      // expect SPI_HOST to send another 1024 bytes to fill RX SRAM FIFO
+      exp_spi_device_irq_rx_above_level = true;
+      exp_spi_device_irq_rx_full = true;
+      exp_spi_device_irq_tx_underflow = true;
+    }
+
+    // after RX SRAM FIFO full, expect RX async FIFO overflow
+    if (spi_device_irq_rx_full_fired) {
+      exp_spi_device_irq_rx_overflow = true;
+      LOG_INFO("SPI_DEVICE RX_FIFO full.");
+    }
+
+    // Wait for the next interrupt to arrive.
```

Comment:
This comment is also redundant. Good comments should say "why", not "what". (Consider removing other comments in this function that are merely describing what code is doing).